### PR TITLE
[MM-22344] Remove ChannelPeek

### PR DESCRIPTION
**Pull request by chuttam**
#### Summary

Current functionality no longer works on iPhone 11 devices, with Haptic Touch replacing/subsuming the previous 3D Touch functionality on iPhones. Current functionality still works on iPhone 8 devices.

Large number of unnecessary re-renders when opening channels, and ideally should not be a part of the channel switching/opening code path. Although this was discovered while trying to investigate an [Android-specific issue](https://mattermost.atlassian.net/browse/MM-22253), this extra code path made it difficult to see what Android is potentially doing differently than iOS.

Originally introduced in #1203. Conversation for removal is [here](https://community.mattermost.com/core/pl/hfcogf6pr7rw8k3ryq14c69c7e).

#### Ticket
[MM-22344](https://mattermost.atlassian.net/browse/MM-22344)

#### Device Information
This PR was tested on: iPhone 11 simulator & device.

### CODE REVIEW

**Comment by chuttam** [comment]
@esethna Can you confirm milestone for me please? While it stemmed from [MM-22253](https://mattermost.atlassian.net/browse/MM-22253), that ticket is still ongoing and we can't be sure it'll make 1.28.

File: app/components/sidebars/main/channels_list/channel_item/channel_item.js [hunk 1]
```diff
@@ -193,11 +170,9 @@ export default class ChannelItem extends PureComponent {
 
         return (
             <AnimatedView>
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/channel_item/channel_item.js]: I guess this is no longer needed

File: assets/base/i18n/en.json [hunk 2]
```diff
@@ -183,7 +183,6 @@
   "mobile.channel_list.not_member": "NOT A MEMBER",
   "mobile.channel_list.unreads": "UNREADS",
   "mobile.channel_members.add_members_alert": "You must select at least one member to add to the channel.",
-  "mobile.channel.markAsRead": "Mark As Read",
```
Comment (enahum) [inline assets/base/i18n/en.json]: ????

File: app/components/sidebars/main/main_sidebar.js [hunk 3]
```diff
@@ -45,7 +45,6 @@ export default class ChannelSidebar extends Component {
         currentUserId: PropTypes.string.isRequired,
         teamsCount: PropTypes.number.isRequired,
         theme: PropTypes.object.isRequired,
-        previewChannel: PropTypes.func,
```
Comment (enahum) [inline app/components/sidebars/main/main_sidebar.js]: This prop is being passed from the Channel screen, we should remove that as well

**Comment by chuttam** [comment]
> ChannelItem needs to remove functionality as well as the ChannelPeek screen needs to be removed from the screens

Screen was already removed. `ChannelItem` functionality removal triggered a cascade of prop deletions for `previewChannel` on many key components. So added QA review just to be certain, since the scope of the change grew.

File: app/components/sidebars/main/channels_list/channel_item/channel_item.js (hunk 1 above)
Comment (chuttam) [reply app/components/sidebars/main/channels_list/channel_item/channel_item.js]: Removed.

File: assets/base/i18n/en.json (hunk 2 above)
Comment (chuttam) [reply assets/base/i18n/en.json]: Only place this was being used was the channel peek preview popup.

File: app/components/sidebars/main/main_sidebar.js (hunk 3 above)
Comment (chuttam) [reply app/components/sidebars/main/main_sidebar.js]: Yes.... `previewChannel` was being passed all through the component tree across several key components. Handled in commit.

File: app/screens/channel/channel.ios.js [hunk 4]
```diff
@@ -25,14 +24,6 @@ const CHANNEL_POST_TEXTBOX_CURSOR_CHANGE = 'onChannelTextBoxCursorChange';
 const CHANNEL_POST_TEXTBOX_VALUE_CHANGE = 'onChannelTextBoxValueChange';
 
 export default class ChannelIOS extends ChannelBase {
-    previewChannel = (passProps, options) => {
-        const screen = 'ChannelPeek';
-
-        peek(screen, passProps, options);
-    };
-
-    optionalProps = {previewChannel: this.previewChannel};
```
Comment (enahum) [inline app/screens/channel/channel.ios.js]: optional props is no longer used,  so it does not need to be included in line 78. Also there is no need for it here https://github.com/mattermost/mattermost-mobile/blob/master/app/screens/channel/channel_base.js#L271 and here https://github.com/mattermost/mattermost-mobile/blob/master/app/screens/channel/channel_base.js#L321

Comment (chuttam) [reply app/screens/channel/channel.ios.js]: Great catch! Done.

**Pull request merged**